### PR TITLE
Feature upgrade

### PR DESCRIPTION
**Pull request by Minglee01**
@incuna/js 

please do a final review. Merging feature upgrade into master.

### CODE REVIEW

File: src/controllers/auth-login-controller.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+(function (angular) {
+    'use strict';
+
+    var module = angular.module('angular-token-auth.auth-login', []);
```
Comment (grahamgilchrist) [inline src/controllers/auth-login-controller.js]: I know we should have picked this up before, but I think these modules should be named more descriptively and consistently with the file names. e.g. `angular-token-auth.auth-login-controller`

File: src/providers/auth-factory.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+(function (angular) {
+    'use strict';
+
+    var module = angular.module('angular-token-auth.auth', [
```
Comment (grahamgilchrist) [inline src/providers/auth-factory.js]: I feel like this module name is too generic, but not sure what else to call it? `angular-token-auth.auth-factory`?

**Comment by grahamgilchrist** [comment]
Looks good. Some comments about naming, but I don't think those are super critical for the merge :)

File: src/controllers/auth-login-controller.js (hunk 1 above)
Comment (Minglee01) [inline src/controllers/auth-login-controller.js]: i remember you telling me before not to use ` -directive` `-controller` etc

**Comment by Minglee01** [comment]
PLease dont merge this. I cant actually get eslint errors.

File: .eslintrc [hunk 3]
```diff
@@ -1,9 +1,19 @@
...
     "rules": {
         "eqeqeq": "error",
         "camelcase": error,
         "curly": error,
         "no-trailing-spaces": error,
+        "wrap-iife": error,
+        "no-use-before-define": error,
+        "new-cap": error,
+        "no-caller": error,
+        "jsx-quotes": error,
```
Comment (grahamgilchrist) [inline .eslintrc]: I think this needs to be the "qoutes" rule: http://eslint.org/docs/rules/quotes

File: .eslintrc [hunk 4]
```diff
@@ -11,7 +11,7 @@
         "no-use-before-define": error,
         "new-cap": error,
         "no-caller": error,
-        "jsx-quotes": error,
+        "quotes": error,
         "strict": error,
         "no-unused-vars": warn,
         "quotes": ["error", "single"]
```
Comment (grahamgilchrist) [inline .eslintrc]: Can we add the `avoidEscape` `["error", "single", {"avoidEscape": true}]`

**Comment by grahamgilchrist** [comment]
Good to merge when tests pass

**Comment by grahamgilchrist** [comment]
👍

**Pull request merged**